tunnel/session.go

```diff
@@ -52,6 +63,10 @@ type message struct {
 	outbound bool
 	// loopback marks the message intended for loopback
 	loopback bool
+	// whether to send as multicast
+	multicast bool
```

Comment:
Should we turn `{multi,broad,uni}cast`s into some enum type rather than keeping them in separate fields?

Reply:
In the future potentially but I've done it this way because I want the ability to internally broadcast without allowing the caller to do so. IPv6 has removed broadcast for appropriate reasons. Its probably best if we avoid it to considering the scale the network could get to.